cabal-install/Distribution/Client/Setup.hs

```diff
@@ -2233,8 +2233,8 @@ parseDependencyOrPackageId = parse Parse.+++ liftM pkgidToDependency parse
   where
     pkgidToDependency :: PackageIdentifier -> Dependency
     pkgidToDependency p = case packageVersion p of
-      Version [] _ -> Dependency (packageName p) anyVersion
-      version      -> Dependency (packageName p) (thisVersion version)
+      v | nullVersion v -> Dependency (packageName p) anyVersion
```

Comment:
this would be just as clear with `v == nullVersion`. So I think we could simplify.